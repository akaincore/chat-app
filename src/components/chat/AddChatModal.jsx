import React from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import Button from '@material-ui/core/Button';

const AddChatModal = ({
 modalOpen, handleModalClose, title, onTitleChange, handleCreateChat,
}) => (
  <Dialog open={modalOpen} onClose={handleModalClose} aria-labelledby="form-dialog-title">
    <DialogTitle id="form-dialog-title">Create new chat</DialogTitle>
    <DialogContent>
      <TextField
        margin="dense"
        name="title"
        label="Chat title"
        type="text"
        required
        fullWidth
        value={title}
        onChange={onTitleChange}
      />
    </DialogContent>
    <DialogActions>
      <Button onClick={handleCreateChat} color="primary">
        Create
      </Button>
      <Button onClick={handleModalClose} color="primary">
        Cancel
      </Button>
    </DialogActions>
  </Dialog>
);

export default AddChatModal;
