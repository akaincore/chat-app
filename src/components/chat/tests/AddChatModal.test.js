/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import AddChatModal from '../AddChatModal';

const mockProps = {
  modalOpen: false,
  handleModalClose: jest.fn(),
  title: 'title',
  onTitleChange: jest.fn(),
  handleCreateChat: jest.fn(),
};

describe('<AddChatModal />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddChatModal {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AddChatModal {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
