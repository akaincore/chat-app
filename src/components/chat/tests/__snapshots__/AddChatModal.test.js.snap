// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddChatModal /> renders correctly 1`] = `null`;
